scheduler/kubernetes/snaps-group.md

```diff
@@ -0,0 +1,227 @@
+---
+layout: page
+title: "Create and use group snapshots"
+keywords: portworx, container, Kubernetes, storage, Docker, k8s, flexvol, pv, persistent disk, snapshots, stork, clones
+sidebar: home_sidebar
+meta-description: "Learn to take a group snapshot of a volume from a Kubernetes persistent volume claim (PVC) and use that snapshot as the volume for a new pod. Try today!"
+---
+
+* TOC
+{:toc}
+
+This document will show you how to create group snapshots of Portworx volumes and how you can clone those snapshots to use them in pods.
+
+## Pre-requisites
+
+**Installing STORK**
+
+{% include k8s/stork/stork-prereq.md %}
+
+**PX Version**
+
+Group snapshots are supported in Portworx version 1.4 and above.
+
+## Taking group snapshots
+
+To take group snapshots, one either specifies annotations that will match PVCs for the group or a Portworx volume group ID.
+
+The group snapshot method supports the following annotations:
+* __px/snapshot-type__: Indicates the type of snapshot. For group snapshots, the value should be **group**.
+* __Annotations to select volumes__: Portworx will select PVCs that match the annotations specified.
+* __px/group-id__: Group ID of the Portworx volumes if they were created using the `--group` parameter. Portworx will select all volumes that match this group ID.
+
+**Example**
+
+In below example, we are taking a group snapshot that will snap all PVCs in the *default* namespace and that have labels *tier: prod* and *type: db*.
+
+Portworx will quiesce I/O on all volumes before triggering their snapshots.
+
+```
+apiVersion: volumesnapshot.external-storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: mysql-snapshot
+  namespace: default
+  annotations:
+    px/snapshot-type: group
+    tier: prod
+    type: db
+spec:
+  persistentVolumeClaimName: mysql-data
+```
+
+*persistentVolumeClaimName* in the above spec is not relevant for group snapshots. It needs to be present in the spec since the schema requires that. The actual PVCs are actually selected using the annotations.
+
+Once you apply the above object you can check the status of the snapshots using `kubectl`:
+
+```
+$ kubectl get volumesnapshot
+NAME                                              AGE
+volumesnapshots/mysql-data-1-779368893912016693   14s
```

Comment:
Why are there 3 snapshot objects? The example should show the list of PVCs and mention which ones will be selected because of the provided selectors.

Reply:
We should delete the original snapshot and name the other ones as <snapshot_name>-<PVC_name>, having 3 objects when there are only 2 snaps seems confusing.